src/lib/comms/include/sol-http-server.h

```diff
@@ -204,10 +214,97 @@ int sol_http_server_set_last_modified(struct sol_http_server *server, const char
  * and parameters (e.g http headers)
  *
  * @return @c 0 on success, error code (always negative) otherwise.
+ *
+ * @see sol_http_server_send_response_progressive
  */
 int sol_http_server_send_response(struct sol_http_request *request, struct sol_http_response *response);
 
 /**
+ * @brief Set the necessary headers to allow server sent events
+ *
+ * @param response The response to set the headers
+ *
+ * @return @c 0 on success, error code (always negative) otherwise.
+ */
+static inline int
+sol_http_response_set_progressive_headers(struct sol_http_response *response)
+{
+    if (!response)
+        return -EINVAL;
```

Comment:
Why not `SOL_NULL_CHECK`? Is it expected to be common flow to call this with no `response`?